### PR TITLE
Implement version string comparison

### PR DESCRIPTION
**Pull request by dsantosmerino**
Similar to https://github.com/growthbook/growthbook/pull/1286, and based on:

Condition | Comparison | Description
-- | -- | --
$veq | `==` | Versions are equal
$vne | `!=` | Versions are not equal
$vlt | `<` | The first version is lesser than the second version
$vlte | `<=` | The first version is lesser than or equal to the second version
$vgt | `>` | The first version is greater than the second version
$vgte | `>=` | The first version is greater than or equal to the second version

Rules:

- Segments are separated by `.` and `-` characters
- Segments should be compared alphanumerically from the left-most segment
  - Digit-only segments should be left-padded with a space so that they have the same number of characters.
- A leading v in a version string should be ignored
- Semantic version syntax used to denote build information (as denoted by a `+`, e.g. `+mybuild`) should be ignored for comparisons.

### CODE REVIEW

**Comment by dsantosmerino** [comment]
Tests will fail until #22 is merged.

**Comment by tinahollygb** [comment]
Looks like you may be developing against an older version of the codebase because you have some conflicts. I can review once resolved.

File: lib/growthbook/conditions.rb [hunk 1]
```diff
@@ -162,6 +174,23 @@ def self.eval_operator_condition(operator, attribute_value, condition_value)
...
+      parts = input.gsub(/(^v|\+.*$)/, "").split(/[-.]/)
+
+      # If it's SemVer without a pre-release, add `~` to the end
+      # ["1","0","0"] -> ["1","0","0","~"]
+      # "~" is the largest ASCII character, so this will make "1.0.0" greater than "1.0.0-beta" for example
+      parts << "~" if(parts.length == 3)
+
+      # Left pad each numeric part with 0s so string comparisons will work ("9">"10", but "09"<"10")
+      parts.map do |part|
+        part.match(/^[0-9]+$/) ? part.rjust(5, "0") : part
```
[the remark this answers is not in the document]
Comment (tinahollygb) [reply lib/growthbook/conditions.rb:190]: I had the same question a while back and was told to use spaces, so let's keep it consistent with the other SDK's please.

[the remark this answers is not in the document]
Comment (dsantosmerino) [reply lib/growthbook/conditions.rb]: Sure, fixed @tinahollygb 😸

**Pull request merged**